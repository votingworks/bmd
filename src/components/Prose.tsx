import styled from 'styled-components'

interface Props {
  textCenter?: boolean
}

const Prose = styled('div')<Props>`
  line-height: 1.2;
  max-width: 41rem;
  padding: 1rem 0;
  text-align: ${({ textCenter }) => (textCenter ? 'center' : undefined)};
  margin: ${({ textCenter }) => (textCenter ? 'auto' : undefined)};
  & h1 {
    margin: 2rem 0 1rem;
    padding: 0 0 1rem;
  }
  & h2 {
    margin: 1.5rem 0 1rem;
  }
  & h3,
  & p {
    margin: 1rem 0;
  }
  & h1 + h2 {
    margin-top: -0.75rem;
  }
  & h1 + p,
  & h2 + p,
  & h3 + p {
    margin-top: -0.75rem;
  }
  & :first-child {
    margin-top: 0;
  }
  & :last-child {
    margin-bottom: 0;
  }
  & div {
    display: grid;
    grid-auto-rows: 1fr;
    grid-gap: 0.5rem;
    align-items: center;

    & p {
      margin: 0;
    }
  }
`

export default Prose
