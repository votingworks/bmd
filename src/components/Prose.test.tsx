import { axe } from 'jest-axe'
import React from 'react'
import { render } from 'react-testing-library'

import Prose from './Prose'

it(`renders Prose`, async () => {
  const { container } = render(
    <Prose>
      <h1>Heading 1</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis
        quam modi magnam neque. Molestias recusandae, officia maiores nam
        pariatur earum qui inventore minus enim adipisci nemo voluptate at
        harum?
      </p>
      <h2>Heading 2</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis
        quam modi magnam neque. Molestias recusandae, officia maiores nam
        pariatur earum qui inventore minus enim adipisci nemo voluptate at
        harum?
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis
        quam modi magnam neque. Molestias recusandae, officia maiores nam
        pariatur earum qui inventore minus enim adipisci nemo voluptate at
        harum?
      </p>
      <h3>Heading 3</h3>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis
        quam modi magnam neque. Molestias recusandae, officia maiores nam
        pariatur earum qui inventore minus enim adipisci nemo voluptate at
        harum?
      </p>
    </Prose>
  )
  expect(await axe(container.innerHTML)).toHaveNoViolations()
  expect(container.firstChild).toMatchSnapshot()
})
