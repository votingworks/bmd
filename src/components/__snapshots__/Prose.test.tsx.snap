// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Prose defaults 1`] = `
.c0 {
  width: 100%;
  max-width: 66ch;
  line-height: 1.2;
}

.c0 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c0 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c0 h3,
.c0 p {
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1rem;
}

.c0 h1 + h2 {
  margin-top: -0.75rem;
}

.c0 h1 + p,
.c0 h2 + p {
  margin-top: -0.75rem;
}

.c0 h3 + p {
  margin-top: -1rem;
}

.c0 > :first-child {
  margin-top: 0;
}

.c0 > :last-child {
  margin-bottom: 0;
}

.c0 dl {
  margin: 1rem 0;
}

@media (min-width:480px) {
  .c0 {
    line-height: 1.3;
  }
}

<div
  class="c0"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;

exports[`renders Prose with compact spacing 1`] = `
.c0 {
  width: 100%;
  max-width: 66ch;
  line-height: 1.2;
}

.c0 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c0 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c0 h3,
.c0 p {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 1rem;
}

.c0 h1 + h2 {
  margin-top: -0.75rem;
}

.c0 h1 + p,
.c0 h2 + p {
  margin-top: -0.75rem;
}

.c0 h3 + p {
  margin-top: 0;
}

.c0 > :first-child {
  margin-top: 0;
}

.c0 > :last-child {
  margin-bottom: 0;
}

.c0 dl {
  margin: 1rem 0;
}

@media (min-width:480px) {
  .c0 {
    line-height: 1.3;
  }
}

<div
  class="c0"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
   
</div>
`;

exports[`renders Prose with compact spacing 2`] = `
.c0 {
  margin: auto;
  width: 100%;
  max-width: 66ch;
  text-align: center;
  line-height: 1.2;
}

.c0 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c0 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c0 h3,
.c0 p {
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1rem;
}

.c0 h1 + h2 {
  margin-top: -0.75rem;
}

.c0 h1 + p,
.c0 h2 + p {
  margin-top: -0.75rem;
}

.c0 h3 + p {
  margin-top: -1rem;
}

.c0 > :first-child {
  margin-top: 0;
}

.c0 > :last-child {
  margin-bottom: 0;
}

.c0 dl {
  margin: 1rem 0;
}

@media (min-width:480px) {
  .c0 {
    line-height: 1.3;
  }
}

<div
  class="c0"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;
